cmake/modules/CheckSubmodules.cmake

```diff
@@ -45,20 +49,42 @@ SET(ENV{LANG} "en_US")
 STRING(REGEX MATCHALL "path = [-0-9A-Za-z/]+" SUBMODULE_LIST ${SUBMODULE_DATA})
 STRING(REGEX MATCHALL "url = [.:%-0-9A-Za-z/]+" SUBMODULE_URL_LIST ${SUBMODULE_DATA})
 
+SET(SKIP_COUNT 0)
 FOREACH(_part ${SUBMODULE_LIST})
 	STRING(REPLACE "path = " "" SUBMODULE_PATH ${_part})
 
 	LIST(FIND SUBMODULE_LIST ${_part} SUBMODULE_INDEX)
 	LIST(GET SUBMODULE_URL_LIST ${SUBMODULE_INDEX} _url)
 	STRING(REPLACE "url = " "" SUBMODULE_URL ${_url})
 
-	# Remove submodules from validation as specified in -DSKIP_SUBMODULES=foo;bar
 	SET(SKIP false)
+	SET(SKIP_REASON "(via SKIP_SUBMODULES)")
+	# Remove unwanted submodules from validation by comparing against -DPLUGIN_LIST=foo;bar
+	IF(${SUBMODULE_PATH} MATCHES "^plugins/")
+		SET(REMOVE_PLUGIN true)
+		FOREACH(_plugin ${PLUGIN_LIST})
+			IF(_plugin STREQUAL "")
+				CONTINUE()
+			ENDIF()
+			IF(${SUBMODULE_PATH} MATCHES "${_plugin}")
+				SET(REMOVE_PLUGIN false)
+			ENDIF()
+		ENDFOREACH()
+
+		IF(REMOVE_PLUGIN)
+			SET(SKIP_REASON "(absent in PLUGIN_LIST)")
+			LIST(APPEND SKIP_SUBMODULES ${SUBMODULE_PATH})
+		ENDIF()
+	ENDIF()
+
+	# Remove submodules from validation as specified in -DSKIP_SUBMODULES=foo;bar
 	IF(SKIP_SUBMODULES)
 		FOREACH(_skip ${SKIP_SUBMODULES})
 			IF(${SUBMODULE_PATH} MATCHES ${_skip})
-				MESSAGE("-- Skipping ${SUBMODULE_PATH} matches \"${_skip}\"")
+				MESSAGE("-- Skipping ${SUBMODULE_PATH} matches \"${_skip}\" ${SKIP_REASON}")
 				SET(SKIP true)
+				MATH(EXPR SKIP_COUNT "${SKIP_COUNT}+1")
+				BREAK()
 			ENDIF()
 		ENDFOREACH()
 	ENDIF()
```

Comment:
Yeah, something's very wrong here.  At a glance, it looks like I'm removing the "_part" entry, which has the `path = `, yet adding the `SUBMODULE_PATH` at the same index.   I'm sure this made sense when I first wrote the loop but now that we're removing items, it really should live as its own, dedicated list.

It may be safe, but it's not readable.

Reply:
Rewritten via 051810b.  This touched a lot of lines, so we'll need to fully retest.